### PR TITLE
feat(transactions): `send_(legacy|eip1559|eip4844)_tx`

### PR DESCRIPTION
**Pull request by zerosnacks**
## Motivation

Requirement for https://github.com/alloy-rs/book/issues/1

## Solution

Adds explicit `send_*_tx` examples for `legacy`, `eip1559` and `eip4844` for clarity

Updates 4844 example to match formatting

## PR Checklist

- [x] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: examples/transactions/examples/send_eip1559_tx.rs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        .with_chain_id(anvil.chain_id())
+        .with_value(U256::from(100))
+        .with_gas_limit(21_000)
+        .with_max_priority_fee_per_gas(1_000_000_000)
+        .with_max_fee_per_gas(20_000_000_000);
+
+    // Send the transaction and wait for the receipt.
+    let pending_tx = provider.send_transaction(tx).await?;
+
+    println!("Pending transaction...{:?}", pending_tx.tx_hash());
```
Comment (DaniPopes) [inline examples/transactions/examples/send_eip1559_tx.rs:40]: B types are Display

```suggestion
    println!("Pending transaction... {}", pending_tx.tx_hash());
```

File: examples/transactions/examples/send_eip1559_tx.rs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    // Wait for the transaction to be included.
+    let receipt = pending_tx.get_receipt().await?;
+
+    println!(
+        "Transaction included in block: {:?}",
+        receipt.block_number.expect("Failed to get block number").to_string()
+    );
```
Comment (DaniPopes) [inline examples/transactions/examples/send_eip1559_tx.rs:48]:
```suggestion
    println!(
        "Transaction included in block {}",
        receipt.block_number.expect("Failed to get block number")
    );
```

**Review by onbjerg — APPROVED** [review]

File: examples/transactions/examples/send_eip4844_tx.rs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        .with_nonce(0)
+        .with_max_fee_per_blob_gas(gas_price)
+        .with_max_fee_per_gas(eip1559_est.max_fee_per_gas)
+        .with_max_priority_fee_per_gas(eip1559_est.max_priority_fee_per_gas)
+        .with_blob_sidecar(sidecar);
+
+    // Send the transaction and wait for the receipt.
+    let pending_tx = provider.send_transaction(tx).await?;
+
+    println!("Pending transaction...{:?}", pending_tx.tx_hash());
```
Comment (onbjerg) [inline examples/transactions/examples/send_eip4844_tx.rs:46]:
```suggestion
    println!("Pending transaction... {}", pending_tx.tx_hash());
```

File: examples/transactions/examples/send_legacy_tx.rs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        .with_to(bob)
+        .with_nonce(0)
+        .with_value(U256::from(100))
+        .with_gas_price(20_000_000_000)
+        .with_gas_limit(21_000);
+
+    // Send the transaction and wait for the receipt.
+    let pending_tx = provider.send_transaction(tx).await?;
+
+    println!("Pending transaction...{:?}", pending_tx.tx_hash());
```
Comment (onbjerg) [inline examples/transactions/examples/send_legacy_tx.rs:38]:
```suggestion
    println!("Pending transaction... {}", pending_tx.tx_hash());
```

File: examples/transactions/examples/send_eip4844_tx.rs (hunk 3 above)
Comment (zerosnacks) [reply examples/transactions/examples/send_eip4844_tx.rs]: thanks!

**Pull request merged**